pyop2/base.py

```diff
@@ -3778,9 +3778,10 @@ def __init__(self, code, name, opts={}, include_dirs=[], headers=[],
             self._code = self._ast_to_c(self._ast, opts)
             search = FindInstances(ast.FunDecl, ast.FlatBlock).visit(self._ast)
             fundecls, flatblocks = search[ast.FunDecl], search[ast.FlatBlock]
-            assert len(fundecls) == 1, "Illegal Kernel"
+            assert len(fundecls) >= 1, "Illegal Kernel"
+            index = [i for i, fundecl in enumerate(fundecls) if fundecl.name == self._name][0]
```

Comment:
`[0]` at end is ugly, because if the condition is satisfied for more than one function declarations then the first is taken and the rest ignored, and I guess this case is not expected at all, so it is better to fail if it still does. Moreover, it is not really the index that you need later, but the function declaration itself, so I would write this line like this:
```python
fundecl, = [fd for fd in fundecls if fd.name == self._name]
```